lib/infra/remote-store-resources.ts

```diff
@@ -0,0 +1,27 @@
+import { RemovalPolicy, Stack } from 'aws-cdk-lib';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import { Effect, Policy, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+
+export class RemoteStoreResources {
+    private readonly snapshotS3Bucket: Bucket
+
+    private readonly bucketPolicyStatement: PolicyStatement
+
+    constructor(scope: Stack) {
+      this.snapshotS3Bucket = new Bucket(scope, `remote-store-${scope.stackName}`, {
+        removalPolicy: RemovalPolicy.DESTROY,
+        autoDeleteObjects: true,
+        bucketName: `${scope.stackName}`,
+      });
+
+      this.bucketPolicyStatement = new PolicyStatement({
+        effect: Effect.ALLOW,
+        actions: ['s3:*'],
```

Comment:
Is this too open for the bucket?

Reply:
Makes sense, updated bucket policy as per `repository-s3` plugin recommendation.